### ISSUE TITLE
Add convenience skip-missing methods for Base aggregation functions

### ISSUE THREAD

**Issue by CameronBieganek** [JuliaData/Missings.jl#146]
xref: https://discourse.julialang.org/t/why-are-missing-values-not-ignored-by-default/106756

There are some users who dislike `mean(skipmissing(x))` because they view it as too verbose or too many keystrokes. Aggregation functions that skip missing values are very commonly needed, so I think it would be worth adding convenience skip-missing methods for Base aggregation functions to Missings.jl, e.g.

- `smean`
- `ssum`
- `svar`
- `scor`

The `svar` and `scor` functions could have a keyword argument to control exactly how missing values are skipped, similar to [`cor` in R](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor).

**Comment by bkamins** [comment]
x-ref https://github.com/JuliaData/DataFrames.jl/issues/3398

**Comment by nalimilan** [comment]
I'm not too much of a fan of these ad-hoc solutions, as the list of functions to support is quite arbitrary. If we go that way we should probably provide functions for all Statistics and StatsBase functions at least, and be prepared to get request to add more functions.

Also the problem is more general than these functions. The annoyance when working with missing values is also to be able to propagate them when transforming values or vectors.

**Comment by CameronBieganek** [comment]
> The annoyance when working with missing values is also to be able to propagate them when transforming values or vectors.

Isn't that already handled by `passmissing`? (I write Python for my day job, so I'm not up to speed on the nuances of missing handling in Julia and DataFrames.)

> If we go that way we should probably provide functions for all Statistics and StatsBase functions at least, and be prepared to get request to add more functions.

I would probably specify in the documentation that only skip-missing version of Base/stdlib functions are provided (and maybe StatsBase??).

**Comment by nalimilan** [comment]
> Isn't that already handled by `passmissing`? (I write Python for my day job, so I'm not up to speed on the nuances of missing handling in Julia and DataFrames.)

Yes, but skipping missing values is already handled by `skipmissing`, so why change anything? :-p My point is that adding lots of functions that skip missing values will only alleviate one part of the problem. A macro in DataFramesMeta would potentially cover most use cases with a single syntax.

**Comment by pdeffebach** [comment]
I think #122 will handle this case... we just need much simpler names. like `skip` or something.

**Comment by knuesel** [comment]
Having a generic solution like @pdeffebach proposes would be nice but I think users would still appreciate simple "direct" functions like `smean` and `ssum` (though maybe `skmean`, `sksum` would be more distinctive while still quite brief).

I think abstract operations like `spreadmissings` add mental load and can be a bit overwhelming for new users who "just want to calculate a mean or a z score". Especially when you have to understand and remember the differences between `skipmissing`, `passmissing`, `skipmissings`, `spreadmissings`...

**Comment by bkamins** [comment]
I have expanded on my thoughts in https://discourse.julialang.org/t/why-are-missing-values-not-ignored-by-default/106756/234
